third_party/dml/ci/test/DispatchTestJobs.ps1

```diff
@@ -47,36 +53,78 @@ $Ado = [ADOHelper]::CreateFromPipeline($AccessToken)
 $TriggerPipeline = $Ado.GetBuild($BuildID)
 
 $Pipeline = $Ado.GetBuildDefinition($TestPipelineName)
+
 $AgentPool = $Ado.GetAgentPool($TestPoolName)
 $AgentQueue = $Ado.GetAgentQueue($TestPoolName)
 $Agents = $Ado.GetAgents($AgentPool.id)
-$DispatchedJobs = [System.Collections.ArrayList]::new()
+
+$WslAgentPool = $Ado.GetAgentPool($WslTestPoolName)
+$WslAgentQueue = $Ado.GetAgentQueue($WslTestPoolName)
+$WslAgents = $Ado.GetAgents($WslAgentPool.id)
+
+$AgentsInfo = [System.Collections.ArrayList]::new()
 
 foreach ($Agent in $Agents)
+{
+    $AgentInfo = 
+    @{
+        'Name' = $Agent.Name;
+        'Status' = $Agent.Status;
+        'Enabled' = $Agent.Enabled;
+        'UserCapabilities' = $Agent.UserCapabilities;
+        'RunOnWsl' = 0;
+        'TestPoolName' = $TestPoolName;
+        'AgentQueueId' = $AgentQueue.id;
+        'Artifacts' = $Artifacts;
+    }
+
+    $AgentsInfo.Add($AgentInfo)
+}
+
+foreach ($WslAgent in $WslAgents)
+{
+    $WslAgentInfo = 
+    @{
+        'Name' = $WslAgent.Name;
+        'Status' = $WslAgent.Status;
+        'Enabled' = $WslAgent.Enabled;
+        'UserCapabilities' = $WslAgent.UserCapabilities;
+        'RunOnWsl' = 1;
+        'TestPoolName' = $WslTestPoolName;
+        'AgentQueueId' = $WslAgentQueue.id;
+        'Artifacts' = $WslArtifacts;
+    }
+
+    $AgentsInfo.Add($WslAgentInfo)
+}
+
+$DispatchedJobs = [System.Collections.ArrayList]::new()
+
+foreach ($AgentInfo in $AgentsInfo)
 {
     $JobInfo = 
```

Comment:
Now that this script dispatches to multiple agent pools, it's possible to have jobs dispatched to two agents with the same name. This will almost certainly break CreateTestEmail.ps1, since it assumes unique agent names. One easy workaround might be embedding the agent pool in the job $JobInfo.AgentName member (e.g. `'AgentName' = "$($AgentInfo.TestPoolName):$($AgentInfo.Name)"`).

Reply:
Oops, sorry about that. I've been having problems with the vscode extension where it doesn't always use the current branch to make the PRs, and I didn't double-check here. I'll re-create this PR but I'll keep your comments in mind for my other branch.